p2panda-rs/src/tests/prop_tests.rs

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use std::convert::TryFrom;
+
+use proptest::prelude::*;
+
+use crate::hash::Hash;
+use crate::operation::{Operation, OperationEncoded, OperationFields};
+
+proptest! {
+    #[test]
+    fn operation_value_as_parameter(fields: OperationFields, hash: Hash) {
+        let operation = Operation::new_create(hash, fields.clone());
+
+        if fields.is_empty() {
+            prop_assert!(operation.is_err());
+        } else {
+            prop_assert!(operation.is_ok());
+
+            let operation_encoded = OperationEncoded::try_from(operation.as_ref().unwrap()).unwrap();
+            let decoded_operation = Operation::from(&operation_encoded);
+
+            prop_assert_eq!(decoded_operation, operation.unwrap());
+        }
+    }
+}
```

Comment:
A module docstring could be nice for this, just something short about what a prop test is.